# Fast and accurate branch lengths estimation for phylogenomic trees | BMC Bioinformatics | Full Text

[https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)

## Abstract

Branch lengths are an important attribute of phylogenetic trees, providing essential information for many studies in evolutionary biology. Yet, part of the current methodology to reconstruct a phylogeny from genomic information — namely supertree methods — focuses on the topology or structure of the phylogenetic tree, rather than the evolutionary divergences associated to it. Moreover, accurate methods to estimate branch lengths — typically based on probabilistic analysis of a concatenated alignment — are limited by large demands in memory and computing time, and may become impractical when the data sets are too large.

Here, we present a novel phylogenomic distance-based method, named ERaBLE (Evolutionary Rates and Branch Length Estimation), to estimate the branch lengths of a given reference topology, and the relative evolutionary rates of the genes employed in the analysis. ERaBLE uses as input data a potentially very large collection of distance matrices, where each matrix is obtained from a different genomic region — either directly from its sequence alignment, or indirectly from a gene tree inferred from the alignment. Our experiments show that ERaBLE is very fast and fairly accurate when compared to other possible approaches for the same tasks. Specifically, it efficiently and accurately deals with large data sets, such as the OrthoMaM v8 database, composed of 6,953 exons from up to 40 mammals.

ERaBLE may be used as a complement to supertree methods — or it may provide an efficient alternative to maximum likelihood analysis of concatenated alignments — to estimate branch lengths from phylogenomic data sets.

## Background

With the continuous growth of genome sequencing capabilities, phylogenetic inference is increasingly based on large collections of genomic regions, if not entire genomes [[1](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)–[3](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]. We have entered the era of phylogenomics — the study of evolution at a genomic scale.

New methodological challenges arise in this field. Clearly, the large amount of data — sequences from several taxa and large collections of genes — makes computational efficiency essential. Besides quantity, the nature of the data is also a concern, and it is extremely important to correctly account for the distinctive features of a typical phylogenomic data set: for example the heterogeneity in the evolution of genomic regions [[4](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)–[9](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)], and the fact that each region is typically sequenced in a subset of the taxa under analysis, with only partial overlap between different subsets [[10](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [11](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)].

In this paper, we focus on the problem of how to efficiently and accurately estimate the branch lengths of a tree in a phylogenomic context, a problem for which, to date, only computationally-intensive techniques appear to be available. Yet, evolutionary distance information is essential to answer several important biological questions, from molecular dating [[12](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [13](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] of events such as speciations, to the reconciliation of gene trees with a species tree [[14](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)], or to the measure of biodiversity in conservation biology [[15](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]. Another goal here is the efficient estimation of the relative rates of evolution of different genomic regions. This information – strictly linked to branch lengths – is also very useful, for example to recognize the diverse selective pressures acting on different parts of the genome [[16](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [17](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]. Tree inferences in a phylogenomic context fall roughly into three frameworks: the supertree, the superalignment and the medium-level framework. We consider them in relation to our goals of branch length and gene rate estimation.

Supertree approaches [[18](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [19](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] combine the information from several phylogenetic trees into a larger phylogeny. A strength of these methods is that the source trees can come from different types of data, such as DNA or protein sequences, or even morphological data. In a phylogenomic context, each source phylogeny is inferred from a different locus, with standard methods such as maximum likelihood, maximum parsimony or distance-based approaches. Within this category, MRP (Matrix Representation with Parsimony) [[20](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [21](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] and its derived methods (e.g., SuperFine [[22](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]) are to date the most widely used approaches. In its standard form, MRP does not use branch length information in the source trees (if present), a limitation that is shared by most supertree methods — with very few exceptions, such as BWD (Build with Distances) [[23](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)], ACS (Average Consensus Supertree) [[24](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] and SDM (Super Distance Matrix) [[25](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]. As a consequence, virtually all supertree approaches are unable to provide meaningful estimates for branch lengths (MRP may provide branch weights, but these should be interpreted as a measure of evidence, not evolutionary change), or any estimate at all for gene rates.

Superalignment methods are the other classical approach for phylogenomic tree inference. They concatenate all available genomic sequence alignments into a unique alignment (often called a character supermatrix), which is then analyzed with standard or specially-tailored phylogenetic reconstruction algorithms [[26](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]. These methods — whose accuracy relies on the use of state-of-the-art statistical inference techniques (typically maximum likelihood or Bayesian methods) — naturally model branch lengths and across-site rate heterogeneity. However they are computationally demanding, and may become impractical if computing time or memory are limited, or when the concatenated alignment is very large. Moreover, heterogeneity in the evolutionary processes at different genomic regions — which is readily handled in a supertree context — may require the use of models such as partition models [[8](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [9](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] or mixture models [[6](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [7](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]. These models, however, further increase the number of parameters to estimate, and consequently computational costs.

Lastly, the medium-level [[25](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [27](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [28](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] framework combines the information originating from the different loci at a level that is intermediate between sequence alignments and complete gene trees. For example, this intermediate level may consist of partial trees — such as quartets [[29](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [30](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] — or pairwise distances between gene sequences [[24](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [25](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]. Specifically, distance-based methods naturally account for and can estimate branch lengths, and in some cases they can even estimate gene rates [[5](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8), [25](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)]. Moreover, they are relatively light computationally. The method we present here, ERaBLE (Evolutionary Rates and Branch Length Estimation), falls within this category. Unlike other phylogenomic methods, however, its goal is not tree estimation, and ERaBLE should be used to complement existing approaches that do not estimate branch lengths and/or gene rates. Alternatively, it can be used on its own when the evolutionary relationships among the species under consideration are largely known.

Note that distance-based phylogenomic methods such as ERaBLE, ACS [[24](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] and SDM [[25](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)] can be used both in the medium-level framework – when the input distances are directly estimated from genomic alignments – but also in the supertree framework – when the input distances require prior inference of a collection of gene trees. We will see examples of this in our experiments (“[Results and discussion](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)” section).

The methodology we propose here can be seen as a generalization of classical weighted least squares (WLS) branch length estimation, to the case where multiple distance matrices are estimated from different genomic regions. In fact if only one matrix is given, ERaBLE behaves exactly like WLS. WLS fits the branch lengths of a tree T so as to make the distances between its leaves as close as possible to the input distances. Formally, it minimises the criterion , where the δ  ij  denote the input distances, the are the distances between the leaves of T (determined by the lengths assigned to its branches), and the weights w  ij >0 express the confidence in the estimate δ  ij . When multiple distance matrices are provided, we face the problem that, due to rate heterogeneity among the alignments, their distances cannot readily be compared to . ERaBLE thus applies a rescaling of the input distances, in order to use them for branch length estimation. Compared to WLS, this entails surprisingly little computational overhead.

In the following, we first describe our new method and the data sets on which we compared its performance to that of other possible approaches for the same task (“[Methods](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)” section). Then, we present the results of our experiments on these data sets (“[Results and discussion](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0821-8)” section).