# Predicting protein-protein interactions using high-quality non-interacting pairs | BMC Bioinformatics | Full Text

[https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)

## Abstract

Identifying protein-protein interactions (PPIs) is of paramount importance for understanding cellular processes. Machine learning-based approaches have been developed to predict PPIs, but the effectiveness of these approaches is unsatisfactory. One major reason is that they randomly choose non-interacting protein pairs (negative samples) or heuristically select non-interacting pairs with low quality.

To boost the effectiveness of predicting PPIs, we propose two novel approaches (NIP-SS and NIP-RW) to generate high quality non-interacting pairs based on sequence similarity and random walk, respectively. Specifically, the known PPIs collected from public databases are used to generate the positive samples. NIP-SS then selects the top-m dissimilar protein pairs as negative examples and controls the degree distribution of selected proteins to construct the negative dataset. NIP-RW performs random walk on the PPI network to update the adjacency matrix of the network, and then selects protein pairs not connected in the updated network as negative samples. Next, we use auto covariance (AC) descriptor to encode the feature information of amino acid sequences. After that, we employ deep neural networks (DNNs) to predict PPIs based on extracted features, positive and negative examples. Extensive experiments show that NIP-SS and NIP-RW can generate negative samples with higher quality than existing strategies and thus enable more accurate prediction.

The experimental results prove that negative datasets constructed by NIP-SS and NIP-RW can reduce the bias and have good generalization ability. NIP-SS and NIP-RW can be used as a plugin to boost the effectiveness of PPIs prediction. Codes and datasets are available at [http://mlda.swu.edu.cn/codes.php?name=NIP](http://mlda.swu.edu.cn/codes.php?name=NIP).

## Background

As the essential component of all organisms, proteins form the very basis of life and carry out a variety of biological functions within living organisms. A protein rarely accomplishes its functions alone, instead it interacts with other proteins to accomplish biological functions. It is thus generally accepted that protein-protein interactions (PPIs) are responsible for most activities of living organisms. As a hotspot of proteomics research, detecting PPIs can not only provide great insight for understanding various biological functions in cells, but also contribute to develop drugs for special diseases[[1](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3), [2](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. In the past decades, different high-throughput technologies had developed to detect PPIs, such as tandem affinity purification (TAP) [[3](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], co-immunoprecipitation (Co-IP) [[4](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], x-ray crystallography [[5](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], yeast two-hybrid (Y2H) screens [[6](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3), [7](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], and mass spectrometric protein complex identification (MS-PCI) [[8](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. However, these wet-experiment based solutions are costly and tedious. PPIs obtained from these biological experiments only cover a small fraction of the complete PPI network [[9](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. Furthermore, these high-throughput technologies generally suffer from high rates of false negatives and false positives [[9](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)–[11](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)].

Computational approaches have been developed to predict PPIs in an economic and reliable way. These approaches use different data types to predict PPIs, such as protein domains [[12](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], protein structure information [[13](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], gene neighborhood [[14](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], gene fusion [[15](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], and phylogenetic profiles [[16](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3), [17](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. Nevertheless, these methods are barely achieved if the pre-knowledge of the proteins is not available, i.e., protein functional domains, 3D structure of proteins, and other information [[18](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. As the explosive growth of sequence data, more and more researchers have moved toward sequence data based approaches to predict PPIs. Experimental results show that it is adequate to predict new PPIs using amino acid sequences alone [[19](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)–[27](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)].

Martin et al. [[19](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)] extracted the feature information of amino acid sequences by the extended signature descriptor and used support vector machine (SVM) to predict PPIs [[19](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. Shen et al. [[20](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)] adopted SVM as the classifier and encoded the feature information of amino acid sequences by conjoint triad (CT), in which the 20 standard amino acids are grouped into 7 categories on the basis of their dipoles and volumes of the side chains. This SVM-based approach yields a high prediction accuracy of 83.9%. However, this approach can not sufficiently encode the feature information, since CT only takes into account the neighboring effect of amino acid sequences, but PPIs usually occur at the non-continuous segments of amino acid sequences. Guo et al. [[21](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)] employed the auto covariance (AC) to detect the correlation among discontinuous segments and obtained an accuracy of 86.55%. You et al. [[24](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)] combined a novel multi-scale continuous and discontinuous (MCD) feature representation and SVM to predict PPIs. MCD feature representation can adequately capture continuous and discontinuous feature information of segments within an amino acid sequence. This method yields a high accuracy of 91.36% [[24](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. Different from these SVM-based approaches, Yang et al. [[22](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)] combined kNN and local descriptor (LD) to predict PPIs and obtained an accuracy of 83.73%. Du et al. [[27](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)] applied deep neural networks (DNNs) and integrated diverse feature descriptors to encode the feature information of amino acid sequences to predict PPIs. This approach obtains a high accuracy of 92.5% on predicting PPIs of Saccharomyces cerevisiae [[27](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. Wang et al. [[28](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)] used DNNs and a novel feature descriptor named local conjoint triad descriptor (LCTD), which encodes continuous and discontinuous feature information of local segments within an amino acid sequence, to predict PPIs. This approach yields a high accuracy of 93.12% on PPIs of Saccharomyces cerevisiae.

However, the performance of all the aforementioned sequence-based methods heavily depend on the quality of PPIs datasets. Positive examples (interacting protein pairs) are generally chosen based on reliable methods (small scale experiments), interactions confirmed by Y2H [[6](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3), [7](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], Co-IP [[4](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], and other methods; or interactions confirmed by interacting paralogs [[29](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3), [30](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. Therefore, given the public protein-protein interactions databases [[31](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], the positive examples are readily available and can be easily constructed. The difficulty is that there are no ‘gold standard’ of non-interacting protein pairs (negative examples), which contribute to discriminatively predict PPIs. Two kinds of strategies are widely used by previous computational methods [[19](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)–[21](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3), [23](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)–[27](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. The first one randomly pairs proteins and then removes the pairs included in the positive examples [[21](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3), [30](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. The second constructs negative examples based on the subcellular localization of proteins [[23](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3), [25](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)–[27](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. However, these two strategies have limitations and may compromise the prediction performance. The first strategy wrongly takes a large number of positive samples as negative samples, while the second strategy leads to a biased estimation of PPIs prediction [[30](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)].

In this paper, two novel approaches (NIP-SS and NIP-RW) are proposed to improve the performance of PPIs prediction. NIP-SS and NIP-RW separately generate reliable non-interacting pairs (negative dataset) based on sequence similarity and on random walk in the PPIs network. The basic idea of NIP-SS is: given a positive protein pair (i and j), and a protein k, the larger the sequence difference between i and k is, the smaller the probability that k interacts with j (i) is. In addition, we control the degree distribution of selected protein pairs to make it similar as that of the positive dataset. Given a PPI network \mathcal {G} =(\mathcal {V},\mathcal {E}), where \mathcal {V} is the set of proteins, and \mathcal {E} is the set of weighted undirected edges, where the weight reflects the interaction strength between a protein pair, 1 means an interaction, 0 means unknown. The basic idea of NIP-RW is: after a k-steps random walk on \mathcal {G}, if the edge weight between two proteins is larger than 0, there may be an interaction between them; otherwise, there may be no interaction.

To investigate the effectiveness of NIP-SS and NIP-RW, we firstly collected the positive sets from Database of Interacting Proteins (DIP) [[31](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], and separately constructed negative sets using four strategies : 1) NIP-SS, 2) NIP-RW, 3) subcellular localization, 4) random pairing, and then merged the positive set and each negative set to form a training dataset. Next, we used the auto covariance (AC) [[21](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)] descriptor to extract the features from amino acid sequences and Deep neural networks (DNNs) to predict PPIs. AC can account for the interactions between residues with a certain distance apart in the sequence and encode the features by a lower dimensional vector [[21](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)], DNNs can automatically extract high-level abstractions and reduce the model training time [[32](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2525-3)]. We performed comparative and quantitative experiments on public benchmark datasets to study the effectiveness of negative datasets generated by different strategies. The experimental results show that NIPI-SS and NIP-RW have good generalization ability and contribute to a higher accuracy in predicting PPIs than other related and widely-used strategies.