# Reading List
  
  - [ ] #interface [[Gainza_et_al_2020_Deciphering_interaction_fingerprints_from_protein_molecular.pdf]]
 - [ ] #proteins [Evolution of resilience in protein interactomes across the tree of life](https://pubmed.ncbi.nlm.nih.gov/30765515/)
 - [ ] #networks [[Devkota_et_al_GLIDER.pdf]]
 - [ ] #covid  [[Fu_et_al_2020_A_Comprehensive_Classification_of_Coronaviruses_and.pdf]]
- [ ]  #proteins [[Jing_Xu_2020_Fast_and_effective_protein_model_refinement_by_deep_graph.pdf]]
- [ ]   #proteins [[Cunningham_et_al_2020_Biophysical_prediction_of_protein–peptide_interactions_and.pdf]]
- [ ]  #proteins #function [Structure-Based Protein Function Prediction using Graph Convlutional Networks](https://www.biorxiv.org/content/10.1101/786236v2.full.pdf)
- [ ] #protiens #embedding #important [[Luo_et_al_Evolutionary_context-integrated_deep_sequence_modeling_for.pdf]]
- [ ] #single-cell #transcriptomics [DRscDB: A single-cell RNA-seq resource for data mining and data comparison across species](https://www.biorxiv.org/content/10.1101/2021.01.29.428862v1)
- [ ] #coral #proteins [Comparitive analysis of the Pocillopora damicornis genome highlights role of immune system in coral evolution](https://www.nature.com/articles/s41598-018-34459-8)
- [ ] #coral [[Okashah et al. - 2019 - Variable G protein determinants of GPCR coupling s.pdf]]
- [ ] #proteins [[AlQuraishi_2019_End-to-End_Differentiable_Learning_of_Protein_Structure.pdf]]
- [x] #proteins #embedding [[Can We Learn the Language of Proteins]] 
- [x] #proteins [[CASP14 what Google DeepMind’s AlphaFold 2 really achieved, and what it means for protein folding, biology and bioinformatics]] 
- [x]  #perturbation [[Interpretable_Machine_Learning_Bo_Yuan.pdf]]
- [x] #proteins [AlphaFold Blog Fabian Fuchs](https://fabianfuchsml.github.io/alphafold2/)
- [x] #proteins [[Porter_et_al_2019_What_method_to_use_for_protein–protein_docking.pdf]]

## NeurIPS
- [x] #neurips #graphs #uncertainty   [[Li_et_al_Stochastic_Deep_Gaussian_Processes_over_Graphs.pdf]] 
- [ ]  #neurips [[You_et_al_Co-Tuning_for_Transfer_Learning.pdf]] 
- [ ]  #neurips  [[Ye_et_al_Towards_Playing_Full_MOBA_Games_with_Deep_Reinforcement.pdf]] 
- [x] #neurips  [[Hie_et_al_Learning_Mutational_Semantics.pdf]]
- [ ]  #neurips #uncertainty  [[Aglietti_et_al_Multi-task_Causal_Learning_with_Gaussian_Processes.pdf]] 
- [x] #neurips #graphs [[Hu_et_al_Open_Graph_Benchmark.pdf]]
- [ ]  #neurips #graphs [[Laenen_Sun_Higher-Order_Spectral_Clustering_of_Directed_Graphs.pdf]]
- [ ]  #neurips [[Maennel_et_al_What_Do_Neural_Networks_Learn_When_Trained_With_Random.pdf]] 
- [ ] #neurips #bayesian #phylogenetics [[Zhang_Improved_Variational_Bayesian_Phylogenetic_Inference_with.pdf]] 
- [ ]  #neurips [[Wang_et_al_A_Combinatorial_Perspective_on_Transfer_Learning.pdf]] 
- [ ]  #neurips #graphs [[Mesquita_et_al_Rethinking_pooling_in_graph_neural_networks.pdf]] 
- [ ] #neurips #graphs #uncertainty [[Zhao_et_al_Uncertainty_Aware_Semi-Supervised_Learning_on_Graph_Data.pdf]]
- [ ] [TrajectoryNet](https://arxiv.org/abs/2002.04461)
- [ ] #neurips #graphs [[Chen_et_al_Iterative_Deep_Graph_Learning_for_Graph_Neural_Networks.pdf]]
- [ ] #neurips #graphs [Subgraph Neural Networks](https://zitniklab.hms.harvard.edu/projects/SubGNN/)
- [ ] #neurips #graphs [[Nikolentzos_Vazirgiannis_Random_Walk_Graph_Neural_Networks.pdf]]
    - [Graph Kernel Method](https://arxiv.org/abs/1911.11119)
- [ ] #neurips #bayesian [[Viinikka_Hyttinen_Towards_Scalable_Bayesian_Learning_of_Causal_DAGs.pdf]]
- [ ] #lmrl #proteins [[Morton_et_al_2020_Protein_Structural_Alignments_From_Sequence.pdf]]
- [ ] #lmrl #perturbation [[Yuan_et_al_Constructing_Data-Driven_Network_Models_of_Cell_Dynamics.pdf]]
- [ ] #lmrl #proteins [[Dallago_et_al_Streamlining_value_of_protein_embeddings_through.pdf]]

# Species Tree Estimation
#covid

- [x] [[Pipes_et_al_Assessing_Uncertainty_in_the_Rooting_of_the_SARS-CoV-2.pdf]]
    - Uses fewer sequences from earlier in the pandemic
    - Compares outgroup rooting with molecular-clock based using ML and Bayesian optimization
- [x] [[Morel_et_al_2020_Phylogenetic_analysis_of_SARS-CoV-2_data_is_difficult.pdf]]
    - Use ~8k sequences available through May
    - Find evidence for a "rugged likelihood surface" --> motivates the use of TreeFix-DTL
    - Recommends against using a single ML tree with default parameters, which we will also support
    - Not very much relevant to use wrt rooting
- [x] [[Sashittal_et_al_2020_Characterization_of_SARS-CoV-2_viral_diversity_within_and.pdf]]

# Recombination Detection
#covid 

- [x] [Recombination should not be an afterthought](https://www.nature.com/articles/s41579-020-00451-1)
    - General motivation for why it is important to study recombination, short article
- [x] [[Pérez-Losada_et_al_2015_Recombination_in_viruses.pdf]]
- [ ] [[Flores-Alanis_et_al_2020_The_receptor_binding_domain_of_SARS-CoV-2_spike_protein_is.pdf]]
- [ ] [[Patiño-Galindo et al. - 2020 - Recombination and lineage-specific mutations led t.pdf]]
- [x] [[Kosakovsky_Pond_et_al_2006_GARD.pdf]]
- [ ] [SimPlot](https://sray.med.som.jhmi.edu/SCRoftware/simplot/)